resolve-workgroup/reports/230926-report.md

```diff
@@ -0,0 +1,31 @@
+# Workgroup Report
+
+**Date:** 2023/09/26
+**Reporter:** Jakub Jezek <jakub@ynput.io>
+
+## Report
+Here are some points which were discussed during the kick off meeting. Order of the points are in preferential order.
+
+### Generating master timeline from multiple timeline variants
+
+To improve NLE editorial support, we can create abstract methods in the back-end that can be used by all integrated NLE supported DCCs. In the DCC project manager, we can select multiple timelines/sequences and use the action "Generate master timeline" to create a new timeline with shared clips in the longest durations. The original timelines should then be duplicated, and the shared clips should point to the new timeline. This is how it is done in Flame, and a similar approach can be achieved in Resolve.
+
+
+### Generating intermediate files for reviewable representations
+
+Similar to Nuke's Bake stream video, we can generate on-the-fly an intermediate video file in Resolve. This file can be then used as input for OIIO and FFMPEG transcoding. It can also have baked color grading look and display & view space.
+
+
+### Colorspace workflow
+
+We can distribute color grading in the same way Hiero does it with soft effects. Users can then load the LUTs to Nuke scripts as Viewer Process Nodes.
+
+
+### Remote transcoding and publishing workflow
+
+To minimize errors in publishing, it would be ideal to locally create a hierarchy, bake intermediate files, and then publish all products and their representations on the farm under the already created folders. Farm publishing can be done via Deadline python jobs.
```

Comment:
Just want to remark that with things like these it's usually best to build the API first so that however the integration down the line happens it uses that API.

E.g.

Define the entry point:

```
transcode timeline.otio
```

Or in Python
```
transcode("timeline.otio')
```

Then all the deadline jobs need to do is just that. But if someone decides they need to run it 'manually' they can also use the API elsewhere, or just use it in the publishing process locally if they really want. Having an easy to use and clear entry point makes (including automated) testing much easier.

In many cases the goals already describe what the end result might be. Take for example a 'feature' that might want to split a video file using an OTIO into its individual files. Then you could prepare an otio file that you know should split in a four video files of a certain length. Then you could start with the test:

```python

def test_split_otio():
    outputs = split("test.otio")
    assert len(outputs) == 4
    assert get_duration(outputs[0]) == 3
    assert get_duration(outputs[1]) == 5
    assert get_duration(outputs[2]) == 10
    assert get_duration(outputs[3]) == 20
```

Then it's a matter of implementing the API method `split` so that it passes the tests and you should be good to go. And then you elaborate the tests based on your goals. What should it do? Test that.

Benefit to having these tests is that it will ensure backwards compatibility (tests always need to pass) and that if a future version of say `OTIO` has some changes in behavior then the tests likely fail and you much quicker detect issues like that to arise.

@fabiaserra has often brought up as well that we should simplify and provide simple API entry points instead of building massive ecosystems straight into the publish plug-in processing because they are hard to then use outside of publishing.

E.g. running a "burnin" similar to how the publishing does it, but doing that manually - I literally have no idea how to do that. There's so many variables that go into it and _that_ appears to be undocumented. But I'd expect it should be something like:

```
result = burnin(source_file, overlays)
```

Where `overlays` could be a list of overlays that need to be drawn on top of the source - with e.g. text per frame.
```
overlays = []
overlays.append(Overlay(text="camera-name", position="topleft", padding=4))
```

Again, I haven't thought to much about that API design. But once you bring it back to the essence of how the burnin works I'd expect you can trivialize it to something like that.

Having the API might also make it easier to swap out the backend with a new system. You'd just need to build it to support the API instead of each new tool getting its own interface/entry points.

Reply:
I definitely think we need a unified pipeline-wide approach. We should talk about this somewhere else, but it's clearly related. Where should we mention this idea? @antirotor, @iLLiCiTiT, @kalisp?

Reply:
so this had been added as goal into AYD-RE-004